### PR TITLE
fix: migrate area chart test to typescript

### PR DESCRIPTION
**Pull request by Yilun-Sun**
## Related Issue

#3126

## Motivation and Context

Migrate AreaChart tests to jest with typescript

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: test-jest/chart/AreaChart.spec.tsx [hunk 1]
```diff
@@ -55,50 +55,37 @@ describe('AreaChart', () => {
     });
   });
 
-  test.skip('Renders customized active dot when activeDot is set to be a ReactElement', () => {
+  test.skip('Renders customized active dot when activeDot is set to be a ReactElement', async () => {
```
Comment (nikolasrieble) [inline test-jest/chart/AreaChart.spec.tsx:58]: Would you like to try to also get the test to work? Or would you prefer to do that in a separate PR?

File: test-jest/chart/AreaChart.spec.tsx [hunk 2]
```diff
@@ -55,50 +55,37 @@ describe('AreaChart', () => {
...
     // eslint-disable-next-line react/prop-types
```
Comment (nikolasrieble) [inline test-jest/chart/AreaChart.spec.tsx:59]: Is this still needed?

File: test-jest/chart/AreaChart.spec.tsx [hunk 3]
```diff
@@ -55,50 +55,37 @@ describe('AreaChart', () => {
...
-    fireEvent.mouseEnter(container, { clientX: 200, clientY: 200 });
-    // fireEvent.mouseEnter(container.querySelector('.recharts-curve'));
+    const area = container.querySelector('.recharts-area-area');
 
-    expect(container.querySelectorAll('.recharts-active-dot')).toHaveLength(1);
-  });
+    if (area !== null) {
+      await userEvent.hover(area);
+      screen.debug(area);
+    }
```
Comment (nikolasrieble) [inline test-jest/chart/AreaChart.spec.tsx:85]: Is the debug needed in the test?
I use it while writing tests, but always remove it in the end.

File: test-jest/chart/AreaChart.spec.tsx (hunk 1 above)
Comment (ckifer) [reply test-jest/chart/AreaChart.spec.tsx:58]: I think this skip was because of the `jsdom` limitations (can't find things on mouse enter/hover

File: test-jest/chart/AreaChart.spec.tsx (hunk 2 above)
Comment (ckifer) [reply test-jest/chart/AreaChart.spec.tsx:59]: +1

File: test-jest/chart/AreaChart.spec.tsx (hunk 3 above)
Comment (ckifer) [reply test-jest/chart/AreaChart.spec.tsx:85]: I think since its in a skip its fine for now

**Review by ckifer — APPROVED** [review]
I think this is fine since we're just moving existing jest tests to TS

We should make a follow up issue to get rid of the skips, make tests better, migrate everything else from `AreaChartSpec.js`, etc. We won't delete `AreaChartSpec.js` until that is done
